### PR TITLE
New ingress work

### PR DESCRIPTION
**Pull request by markyjackson-taulia**

### CODE REVIEW

File: sysdigcloud/sysdigcloud-scanning-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+apiVersion: extensions/v1beta1
```
Comment (papajulio) [inline sysdigcloud/sysdigcloud-scanning-ingress.yaml]: This file it's necessary? What's the difference with scanning-ingress.yaml ??

File: sysdigcloud/sysdigcloud-api-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+apiVersion: extensions/v1beta1
```
Comment (papajulio) [inline sysdigcloud/sysdigcloud-api-ingress.yaml]: Is this file necessary? What's the difference with api-ingress.yaml ??

File: sysdigcloud/haproxy-ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,226 @@
+---
```
Comment (papajulio) [inline sysdigcloud/haproxy-ingress.yaml]: This file can deleted right? You already split this into the yaml's in the ingress-controller folder...

File: sysdigcloud/ingress_controller/ingress-daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      labels:
+        run: haproxy-ingress
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        role: ingress-controller
+      serviceAccountName: ingress-controller
+      containers:
+        - name: haproxy-ingress
+          image: quay.io/sysdig/haproxy-ingress:0.0.1
```
Comment (papajulio) [inline sysdigcloud/ingress_controller/ingress-daemonset.yaml]: To pull from here we need a ImagePullSecret right? Shouldn't we add it in this file?

File: sysdigcloud/ingress_controller/ingress-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  template:
+    metadata:
+      labels:
+        run: ingress-default-backend
+    spec:
+      imagePullSecrets:
+        - name: sysdigcloud-pull-secret
+      containers:
+        - name: ingress-default-backend
+          image: quay.io/sysdig/ingress-default-backend:0.0.1
```
Comment (papajulio) [inline sysdigcloud/ingress_controller/ingress-deployment.yaml]: To pull from here we need a ImagePullSecret right? Shouldn't we add it in this file?

File: sysdigcloud/api-headless-service.yaml [hunk 6]
```diff
@@ -2,16 +2,15 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: sysdigcloud-api
+  name: api-headless-service
```
Comment (papajulio) [inline sysdigcloud/api-headless-service.yaml]: This change needs to be updated in all the yamls that reference this service or be reverted. Or it won't work by default

File: sysdigcloud/collector-headless-service.yaml [hunk 7]
```diff
@@ -2,16 +2,15 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: sysdigcloud-collector
+  name: collector-headless-service
```
Comment (papajulio) [inline sysdigcloud/collector-headless-service.yaml]: This change needs to be updated in all the yamls that reference this service or be reverted. Or it won't work by default

**Comment by papajulio** [comment]
@markyjackson-taulia another side note to the other comments, I'm missing some doc about it somewhere. It will help a lot to have some intructions to follow when applying this.

File: sysdigcloud/haproxy-ingress.yaml (hunk 3 above)
Comment (markyjackson-taulia) [reply sysdigcloud/haproxy-ingress.yaml]: Correct, I forgot to remove that and will do so now

File: sysdigcloud/ingress_controller/ingress-daemonset.yaml (hunk 4 above)
Comment (markyjackson-taulia) [reply sysdigcloud/ingress_controller/ingress-daemonset.yaml]: You are correct

File: sysdigcloud/api-headless-service.yaml (hunk 6 above)
Comment (markyjackson-taulia) [reply sysdigcloud/api-headless-service.yaml]: Reverted

File: sysdigcloud/sysdigcloud-api-ingress.yaml (hunk 2 above)
Comment (markyjackson-taulia) [reply sysdigcloud/sysdigcloud-api-ingress.yaml]: Dup and removed

File: sysdigcloud/sysdigcloud-scanning-ingress.yaml (hunk 1 above)
Comment (markyjackson-taulia) [reply sysdigcloud/sysdigcloud-scanning-ingress.yaml]: Dup and removed

File: sysdigcloud/collector-headless-service.yaml (hunk 7 above)
Comment (mbreitung) [reply sysdigcloud/collector-headless-service.yaml]: this is reverted

File: sysdigcloud/ingress_controller/ingress-daemonset.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          args:
+              - --reload-strategy=multibinder
+              - --default-backend-service=$(POD_NAMESPACE)/default-http-backend
```
Comment (papajulio) [inline sysdigcloud/ingress_controller/ingress-daemonset.yaml]: The service in this yamls is named differently this line should be: ` - --default-backend-service=$(POD_NAMESPACE)/ingress-default-backend ` Without this change the daemonset doesn't start properly...

File: sysdigcloud/collector-headless-service.yaml [hunk 9]
```diff
@@ -7,11 +7,11 @@ metadata:
     app: sysdigcloud
     role: collector
 spec:
-  type: LoadBalancer
+  clusterIP: None
   ports:
     - port: 6443
       name: secure-collector
-      targetPort: 6443
+      targetPort: 6000
```
Comment (papajulio) [inline sysdigcloud/collector-headless-service.yaml]: I have a doubt about this port, we are exposing here the 6000 port, shouldn't be 6443 ?? Or this is ok?

File: sysdigcloud/api-ingress.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      http:
+        paths:
+          - backend:
+              serviceName: sysdigcloud-api
+              servicePort: 443
+            path: /api
+          - backend:
+              serviceName: sysdigcloud-api
+              servicePort: 443
+            path: /ui
```
Comment (papajulio) [inline sysdigcloud/api-ingress.yaml]: Here we are exposing only the /ui and /api endpoints from the API, but in onprem we need to access to / and get the login page that it's serverd by the API, could be that we are re-using here the staging/prod config where the frontend it's served from other part (S3??) ??

File: sysdigcloud/scanning-ingress.yaml [hunk 11]
```diff
@@ -1,27 +0,0 @@
----
```
Comment (papajulio) [inline sysdigcloud/scanning-ingress.yaml]: I think this file was valid, you delete too much this time I think :)

File: sysdigcloud/collector-headless-service.yaml (hunk 9 above)
Comment (markyjackson-taulia) [reply sysdigcloud/collector-headless-service.yaml]: This is ok. @mbreitung and I discussed

File: sysdigcloud/scanning-ingress.yaml (hunk 11 above)
Comment (markyjackson-taulia) [reply sysdigcloud/scanning-ingress.yaml]: opps. I added it back

File: sysdigcloud/collector-headless-service.yaml (hunk 9 above)
Comment (papajulio) [reply sysdigcloud/collector-headless-service.yaml]: @markyjackson-taulia we forgot to add this to our PR, but this change is wrong it doesn't work if you don't leave as it was, with 6443
